libs/testdiff/replacement.go

```diff
@@ -74,13 +76,48 @@ func (r *ReplacementsContext) Set(old, new string) {
 	if err == nil {
 		encodedOld, err := json.Marshal(old)
 		if err == nil {
-			r.appendLiteral(string(encodedOld), string(encodedNew))
+			r.appendLiteral(trimQuotes(string(encodedOld)), trimQuotes(string(encodedNew)))
 		}
 	}
 
 	r.appendLiteral(old, new)
 }
 
+func trimQuotes(s string) string {
```

Comment:
Why is this needed?

If we only care about quoting strings, you can look at `fmt.Sprintf` with `%q`.

Reply:
I care about strings inside JSON that might have been concatenated with others, e.g. `xy` becomes `"XY"` in json. Now I need to replace x -> X.